homework/index.js

```diff
@@ -30,17 +34,96 @@
     return elem;
   }
 
+  function renderError(error) {
+    const root = document.getElementById('root');
+    root.innerHTML = '';
+    createAndAppend('h1', root, { text: error.message, class: `alert-error` });
+  }
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepos(repo, div) {
+    div.innerHTML = '';
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: `_blank`,
+    });
+    addRow(tbody, 'Description', repo.description);
+    addRow(tbody, 'Forks', repo.forks);
+    addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString('en-GB'));
+  }
+
+  function renderContributions(repo, ul) {
+    fetchJSON(repo.contributors_url, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+      ul.innerHTML = '';
+      createAndAppend('h3', ul, { text: `Contributors:` });
+      contributors.forEach(contributor => {
+        const a = createAndAppend('a', ul, { href: contributor.html_url, target: '_blank' });
+        const li = createAndAppend('li', a);
+        const table = createAndAppend('table', li);
+        const tbody = createAndAppend('tbody', table);
+        const tr1 = createAndAppend('tr', tbody);
+        createAndAppend('img', tr1, { src: contributor.avatar_url });
+        createAndAppend('td', tr1, { text: contributor.login });
+        createAndAppend('td', tr1, { text: contributor.contributions });
+      });
+    });
+  }
+
+  function createOptionElements(repositories, select) {
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repo, index) => {
+        createAndAppend('option', select, { text: repo.name, value: index });
+      });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header);
+    const mainContainer = createAndAppend('div', root, { id: `main-container` });
+    const div1 = createAndAppend('div', mainContainer, { id: 'table-container' });
```

Comment:
Try and avoid variable names with (meaningless) number suffixes. You already came up with good, intention-revealing names for the `id` attributes, so why not also use those names in camelCase form as the variable name?

```js
const tableContainer = createAndAppend('div', mainContainer, { id: 'table-container' });
```

Reply:
You are right. I changed the variable names I will be more careful about this in the future